homework3/task04/project_is_armstrong/task4_armstrong.py

```diff
@@ -0,0 +1,30 @@
+"""
+Task.
+
+Armstrong number is a number that is the sum of its own digits each raised to the power of the number of digits.
+https://en.wikipedia.org/wiki/Narcissistic_number
+
+Examples:
+
+- 9 is an Armstrong number, 9 = 9^1 = 9
+- 10 is not: 10 != 1^2 + 0^2 = 1
+- 153 is : 153 = 1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153
+
+
+Write a function that detects if a number is Armstrong number in functionaly style:
+ - use map or other utilities from functools library,
+ - use anonymous functions (or use function as argument)
+ - do not use loops, preferably using list comprehensions
+
+### Example function signature and call
+"""
+from functools import reduce
+
+
+def is_armstrong(number: int) -> bool:
+    """Return True if a number is Armstrong number."""
+    power = len(str(number))
+    result = reduce(
+        lambda x, y: x + (y ** power), list(map(int, (i for i in str(number))))
```

Comment:
Why list?

Reply:
because map() returns map object. But, I tried to remove list and while debugging I saw strange thing.  It works without list, but I cant understand why. So, I can suppose map object is also iterable.
![image](https://user-images.githubusercontent.com/64098113/111776375-fcb9cc80-88c2-11eb-8e54-195209763ef2.png)

Reply:
>  It works without list, but I cant understand why. So, I can suppose map object is also iterable.

There is a more easy way to detect it. If you can pass it to `list()`, it is itearable.